indico/modules/events/papers/operations.py

```diff
@@ -164,3 +168,86 @@ def delete_paper_template(template):
     db.session.delete(template)
     db.session.flush()
     logger.info('Paper template %r deleted by %r', template, session.user)
+
+
+def create_review(paper, type_, user, review_data, questions_data):
+    review = PaperReview(revision=paper.last_revision, type=type_.instance, user=user)
+    review.populate_from_dict(review_data)
+    log_data = {}
+    for question in paper.event_new.cfp.get_questions_from_review_type(type_.instance):
+        value = int(questions_data['question_{}'.format(question.id)])
+        review.ratings.append(PaperReviewRating(question=question, value=value))
+        log_data[question.text] = value
+    db.session.flush()
+    logger.info("Paper %s received a review of type %s by %s", paper, type_, user)
+    log_data.update({
+        'Type': unicode(type_.title),
+        'Action': orig_string(review.proposed_action.title),
+        'Comment': review.comment
+    })
+    paper.event_new.log(EventLogRealm.management, EventLogKind.positive, 'Papers',
+                        'Paper for contribution {} reviewed'.format(paper.contribution.verbose_title),
+                        user, data=log_data)
+    return review
+
+
+def update_review(review, review_data, questions_data):
+    paper = review.revision.paper
+    event = paper.event_new
+    changes = review.populate_from_dict(review_data)
+    review.modified_dt = now_utc()
+    log_fields = {}
+    for question in event.cfp.get_questions_from_review_type(review.type):
+        field_name = 'question_{}'.format(question.id)
+        rating = question.get_review_rating(review, allow_create=True)
+        old_value = rating.value
+        rating.value = int(questions_data[field_name])
+        if old_value != rating.value:
+            changes[field_name] = (old_value, rating.value)
+            log_fields[field_name] = {
+                'title': question.text,
+                'type': 'number'
+            }
+    db.session.flush()
+    logger.info("Paper review %s modified", review)
+    log_fields.update({
+        'proposed_action': 'Action',
+        'comment': 'Comment'
+    })
+    event.log(EventLogRealm.management, EventLogKind.change, 'Papers',
+              'Review for paper {} modified'.format(paper.verbose_title),
+              session.user, data={'Changes': make_diff_log(changes, log_fields)})
+
+
+@no_autoflush
+def create_comment(paper, comment_data, user):
+    comment = PaperReviewComment(user=user)
+    comment.populate_from_dict(comment_data)
+    comment.paper_revision = paper.last_revision
+    db.session.flush()
+    logger.info("Paper %s received a comment from %s", paper, session.user)
+    paper.event_new.log(EventLogRealm.management, EventLogKind.positive, 'Papers',
+                        'Paper {} received a comment'.format(paper.verbose_title),
+                        session.user)
+
+
+def update_comment(comment, comment_data):
+    changes = comment.populate_from_dict(comment_data)
+    comment.modified_by = session.user
```

Comment:
We need to pass the user explicitly since `session.user` resolves to `None` from the CLI. While that's fine for the logger, we don't want to end up with `comment.modified_dt` set to a timestamp while `comment.modified_by` pointing to a ghost user.